src/garage/replay_buffer/replay_buffer.py

```diff
@@ -12,173 +12,75 @@
-        time_horizon (int): time horizon of rollout.
-
+        capacity_in_transitions (int): capacity of the replay buffer,
+            measured in transitions.
     """
 
-    def __init__(self, env_spec, size_in_transitions, time_horizon):
-        del env_spec
+    def __init__(self, env_spec, capacity_in_transitions):
+        self.capacity_in_transitions = capacity_in_transitions
+        self.env_spec = env_spec
         self._current_size = 0
         self._current_ptr = 0
         self._n_transitions_stored = 0
-        self._time_horizon = time_horizon
-        self._size_in_transitions = size_in_transitions
-        self._size = size_in_transitions // time_horizon
-        self._initialized_buffer = False
         self._buffer = {}
-        self._episode_buffer = {}
-
-    def store_episode(self):
-        """Add an episode to the buffer."""
-        episode_buffer = self._convert_episode_to_batch_major()
-        rollout_batch_size = len(episode_buffer['observation'])
-        idx = self._get_storage_idx(rollout_batch_size)
-
-        for key in self._buffer:
-            self._buffer[key][idx] = episode_buffer[key]
-        self._n_transitions_stored = min(
-            self._size_in_transitions, self._n_transitions_stored +
-            self._time_horizon * rollout_batch_size)
 
     @abstractmethod
-    def sample(self, batch_size):
-        """Sample a transition of batch_size.
+    def sample_transitions(self, batch_size):
+        """Sample N transitions, where N=batch_size.
 
         Args:
-            batch_size(int): The number of transitions to be sampled.
+            batch_size (int): The number of transitions to be sampled.
 
+        Returns:
+            SampleBatch: batch_size transitions sampled.
         """
-        raise NotImplementedError
 
-    def add_transition(self, **kwargs):
-        """Add one transition into the replay buffer.
+    def sample_path(self):
+        """Sample one path of arbitrary length.
 
-        Args:
-            kwargs (dict(str, [numpy.ndarray])): Dictionary that holds
-                the transitions.
+        Returns:
+            SampleBatch: Batch of transitions that make up a path.
 
+        Raises:
+            NotYetImplementedError: If the replay buffer does not support
+                path sampling.
         """
-        transition = {k: [v] for k, v in kwargs.items()}
-        self.add_transitions(**transition)
-
-    def add_transitions(self, **kwargs):
-        """Add multiple transitions into the replay buffer.
+        raise NotImplementedError
 
-        A transition contains one or multiple entries, e.g.
-        observation, action, reward, terminal and next_observation.
-        The same entry of all the transitions are stacked, e.g.
-        {'observation': [obs1, obs2, obs3]} where obs1 is one
-        numpy.ndarray observation from the environment.
+    @abstractmethod
+    def add_path(self, path):
```

Comment:
`add_trajectory` since the dtype is `TrajectoryBatch`

consider defining a dtype `Trajectory` if the singleton case appears often.

Reply:
if `Trajectory` dtype exists, this could accept `Union[Trajectory, TrajectoryBatch]`